### PR TITLE
Support for webpack 5

### PR DESCRIPTION
**Pull request by kmalakoff**
I was using this plugin for webpack 4 and wanted to use it for webpack 5.

If you are interested in accepting with pull request, there are a few points of note:

- I republished it under @effortlessmotion/dynamic-cdn-webpack-plugin so we'd need to change the module name back
- I added a line to the README.md warning this is a fork when I published it under a different org
- I made it work with Node 10 like webpack guarantees as a compatible version 
- I removed ava and replaced it with tape as I couldn't get the latest ava to test on Node 10
- I removed babel and es6 modules to make this simpler to test, run, and publish without extra steps. Webpack 5 uses require for backwards compatibility without transpiling

### CODE REVIEW

**Comment by ChocolateLoverRaj** [comment]
@kmalakoff can you please link this to say it fixes #71?

**Comment by kmalakoff** [comment]
This fixes [issue 70](https://github.com/mastilver/dynamic-cdn-webpack-plugin/issues/70)

File: readme.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 > Dynamically get your dependencies from a cdn rather than bundling them in your app
 
+Warning: This module is a fork from [https://github.com/mastilver/dynamic-cdn-webpack-plugin](https://github.com/mastilver/dynamic-cdn-webpack-plugin) to support Webpack 5.
```
Comment (jmfrancois) [inline readme.md:7]: should not be here :)

Comment (mg-83) [reply readme.md:7]: as he said in his comment

**Comment by mg-83** [comment]
@mastilver are you still updating this?

**Comment by aulisius** [comment]
Hi @kmalakoff ! One of the maintainers here :)

Thanks for the huge help in adding Webpack 5 support to this plugin. Much appreciated.

I'm unable to reach @mastilver (I'll try again) so I'm unsure as to how we would publish to npm but I don't think that should stop us from merging this into the codebase.

I'll go through your PR and get back on this asap. If in the event we aren't able to push this to npm, I think we can add a message to the README pointing to the `@effortlessmotion/dynamic-cdn-webpack-plugin` for now.

**Comment by richardantao** [comment]
Hi @aulisius, any updates?

**Comment by aulisius** [comment]
Hi @richardantao, you can use `@effortlessmotion/dynamic-cdn-webpack-plugin` package which supports webpack 5.

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
+  include:
+    # - stage: Tests
+    #   name: Unit Tests on OSX
+    #   os: osx
+    - stage: Tests
+      name: Unit Tests on Linux
+      os: linux
+    - stage: Tests
+      name: Unit Tests on Windows
+      os: windows
```
Comment (elamperti) [inline .travis.yml:15]: What does it add to test on Windows in the context of this plugin?

File: package.json [hunk 3]
```diff
@@ -1,27 +1,29 @@
 {
-  "name": "dynamic-cdn-webpack-plugin",
-  "version": "4.0.0",
+  "name": "@effortlessmotion/dynamic-cdn-webpack-plugin",
```
Comment (elamperti) [inline package.json:2]: This shouldn't change in this PR.

File: package.json [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
+  "version": "5.0.1",
   "description": "Dynamically get your dependencies from a cdn rather than bundling them in your app",
   "license": "MIT",
   "repository": "mastilver/dynamic-cdn-webpack-plugin",
   "author": {
     "name": "Thomas Sileghem",
     "email": "th.sileghem@gmail.com",
-    "url": "mastilver.com"
+    "url": "https://github.com/mastilver"
```
Comment (elamperti) [inline package.json:10]: This is up to the author and irrelevant to the PR

File: src/index.js [hunk 5]
```diff
@@ -1,112 +1,102 @@
...
-let HtmlWebpackPlugin;
+let HtmlWebpackPlugin = null;
 try {
-    // eslint-disable-next-line import/no-extraneous-dependencies
     HtmlWebpackPlugin = require('html-webpack-plugin');
-} catch (err) {
-    HtmlWebpackPlugin = null;
-}
+// eslint-disable-next-line unicorn/prefer-optional-catch-binding
+} catch (_) {}
```
Comment (elamperti) [inline src/index.js:9]: This doesn't improve legibility and I don't see any improvement in performance either.

File: src/index.js [hunk 6]
```diff
@@ -1,112 +1,102 @@
...
     }
 };
 
-export default class DynamicCdnWebpackPlugin {
+module.exports = class DynamicCdnWebpackPlugin {
     constructor({disable = false, env, exclude, only, verbose, resolver} = {}) {
-        if (exclude && only) {
-            throw new Error('You can\'t use \'exclude\' and \'only\' at the same time');
-        }
+        if (exclude && only) throw new Error('You can\'t use \'exclude\' and \'only\' at the same time');
```
Comment (elamperti) [inline src/index.js:28]: This formatting change isn't relevant to the PR (and may be opposed to the style the author wants to use)

File: src/index.js [hunk 7]
```diff
@@ -1,112 +1,102 @@
...
+            return isSameVersion ? this.modulesFromCdn[modulePath].var : false;
         }
 
         const cdnConfig = await this.resolver(modulePath, version, {env});
 
         if (cdnConfig == null) {
-            if (this.verbose) {
-                console.log(`❌ '${modulePath}' couldn't be found, please add it to https://github.com/mastilver/module-to-cdn/blob/master/modules.json`);
-            }
+            if (this.verbose) console.log(`❌ '${modulePath}' couldn't be found, please add it to https://github.com/mastilver/module-to-cdn/blob/master/modules.json`);
```
Comment (elamperti) [inline src/index.js:99]: Same as above, and in this case the `console.log` line is already long so it difficults reading the whole line on small or split screens.

**Review by elamperti — CHANGES_REQUESTED** [review]
@kmalakoff First of all, thanks for taking the time to make all the required changes to update this package! I've noticed some changes that may not be directly related to the PR and some things that could use clarification.

In regard to your fork, it's great to have an alternative! This PR should keep all references to the original package as they are, so if this PR is accepted the changes can be reflected for all users.

It could be useful if @mastilver considers making you (or anyone else who may have the time and knowledge) maintainer of the repo to keep the project alive.

**Comment by jmfrancois** [comment]
FYI we have never been able after multiple attempts to contribute back to this module with module-to-cdn.
We have decided to fork it under Talend github organisation.
The fork is public and we have deployed sites in production with custom CDN based on this fork.
https://github.com/Talend/ui/tree/master/fork

We have added multiple features to answer needs for deployments:
* possibility to have other CDN than unpkg.
* url formating
* add metadata to html webpack plugin
* fix bug around dependencies management
* introduce a dependencies.json file along UMDs

We plan to add webpack 5 support this year

**Comment by cosmincc** [comment]
Would be great to have this done. I'm sure the maintainers could update the pull request and fix the issues directly, no?
Can anyone else help speed this up?

**Comment by aulisius** [comment]
There's no blockers to merge this PR but only @mastilver holds the access to publish on npm. So, even if the PR is merged, it wouldn't be updated on npm

**Comment by Nishchit14** [comment]
@mastilver would be grateful if you can publish it. thank you.

**Comment by CSDev0** [comment]
author is dead or what? this project is saddly no longer maintained as i see.

**Pull request merged**